main.tf

```diff
@@ -0,0 +1,692 @@
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "443"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  resource_group_name         = azurerm_resource_group.resource_group.name
+  network_security_group_name = azurerm_network_security_group.nsg_azbastion.name
+}
+
+# Create NIC.
+resource "azurerm_network_interface" "webTier" {
+  name                = "webTier_nic"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  ip_configuration {
+    name                          = "ipconfiguration_web"
+    subnet_id                     = azurerm_subnet.webTierSubnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+resource "azurerm_network_interface" "webTier2" {
+  name                = "webTier_nic2"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  ip_configuration {
+    name                          = "ipconfiguration_web"
+    subnet_id                     = azurerm_subnet.webTierSubnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+resource "azurerm_network_interface" "businessTier" {
+  name                = "business_nic"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  ip_configuration {
+    name                          = "ipconfiguration_business"
+    subnet_id                     = azurerm_subnet.businessTierSubnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+resource "azurerm_network_interface" "businessTier2" {
+  name                = "business_nic2"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  ip_configuration {
+    name                          = "ipconfiguration_business"
+    subnet_id                     = azurerm_subnet.businessTierSubnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+resource "azurerm_network_interface" "dataTier" {
+  name                = "data_nic"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  ip_configuration {
+    name                          = "ipconfiguration_data"
+    subnet_id                     = azurerm_subnet.dataTierSubnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+resource "azurerm_network_interface" "dataTier2" {
+  name                = "data_nic2"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  ip_configuration {
+    name                          = "ipconfiguration_data"
+    subnet_id                     = azurerm_subnet.dataTierSubnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+# Create VMs for Web Tier.
+resource "azurerm_windows_virtual_machine" "webTierVm1" {

```

Comment:
These VMs are a really good example of where modules would come in handy.  You could move all of the repeated stuff to a module and make variables out of the differences.